native/cocos/3d/assets/Mesh.cpp

```diff
@@ -292,15 +294,26 @@ void Mesh::initialize() {
 
                 uint32_t dstStride = idxView.stride;
                 uint32_t dstSize = idxView.length;
-                if (dstStride == 4 && !gfxDevice->hasFeature(gfx::Feature::ELEMENT_INDEX_UINT)) {
+                if (dstStride == 4) {
                     uint32_t vertexCount = _struct.vertexBundles[prim.vertexBundelIndices[0]].view.count;
-                    if (vertexCount >= 65536) {
-                        debug::warnID(10001, vertexCount, 65536);
+#if CC_OPTIMIZE_MESH_DATA
+                    if (vertexCount < 65536) {
```

Comment:
What happen if `vertexCount >= 65536`?

Reply:
When CC_OPTIOMIZE_MESH_DATA is enabled, if vertexCount >= 65536, there are two situations:
1. GPU support UINT element index type, then do nothing
2. GPU doesn't support UNIT element type, just ignore the primitive (mesh) to be uploaded to GPU.
if vertexCount is < 65536, convert index type to ushort.

When CC_OPTIOMIZE_MESH_DATA is disabled, if vertexCount is < 65536 and GPU doesn't support UINT element index type, converting index type to ushort.